Dispatcher/Repositories/SearchRepository.swift

```diff
@@ -0,0 +1,34 @@
+//
+//  SearchRepository.swift
+//  Dispatcher
+//
+//  Created by Ofek Cohen on 09/05/2023.
+//
+
+import Foundation
+
+class SearchRepository {
+    
+    static let shared = SearchRepository()
```

Comment:
Why is it a singleton? Only the SeachViewModel uses it

Reply:
I did it a singleton because the ArticlesRepository is a singleton and I thought that it should be the same in order to be consistent

Reply:
הסינגלטון הוא פיתרון שימושי למצבים שבהם אנחנו חייבים שהקלאס יהיה זמין ונגיש לנו ברחבי האפליקציה והדאטה שהוא מחזיק יהיה זהה מכל מקום שניגש אליו - זה עונה על קונספט של single source of truth
אבל חיסרון מרכזי של הסינגלטון הוא שברגע שהוא נוצר לראשונה - הוא יחייה לנצח כל עוד האפליקציה בחיים, גם אם כבר אין לנו בו צורך או שימוש, והוא ממשיך לגזול משאבים של זיכרון שעולה לנו בפרפורמנס של האפליקציה.
ולכן אנחנו לא ניצור סתם סינגלטונים, אלא רק אם יש באמת צורך בכך. במקרה פה למשל, רק ה-SearchViewModel צריך להשתמש ב-SearchRepository ולכן אין לנו צורך ליצור ממנו סינגלטון